### ISSUE TITLE
Add terminate and decrement operation

### ISSUE THREAD

**Issue by drmaas** [spinnaker/spinnaker#1187]
I do not see this as feasible using heat.

**Comment by drmaas** [comment]
If we terminate an instance, then scale down the stack, the stack may decide to deallocate a different instance.

If we scale down a stack first, we can't specify which resources to remove.

Perhaps after moving to senlin this will be achievable.

**Comment by drmaas** [comment]
Update - once https://github.com/spinnaker/clouddriver/pull/1111 is merged, this should be simple to implement - it will do the same thing as the terminate operation, but will also decrement the sizes by 1 before issuing the stack update.

**Comment by drmaas** [comment]
This should decrement the min size and the desired size, and potentially also decrement the max size or set it to the new desired size - aws provides an option. It is not clear how the google api behaves.

**Comment by drmaas** [comment]
On hold - it is not clear if the resource marked unhealthy will be the same one removed when decementing the asg count - in fact, it seems likely that openstack would pick an instance at random to remove.

**Comment by lwander** [comment]
Closing because "wont-fix"
